swiftui/PasswordlessSwiftUI/PasswordlessSwiftUI/ContentView.swift

```diff
@@ -0,0 +1,181 @@
+//
+//  ContentView.swift
+//  PasswordlessSwiftUI
+//
+//  Copyright © 2020 Google LLC. All rights reserved.
+//
+
+import SwiftUI
+import Firebase
+
+
+/// Main view where user can login in using Email Link Authentication.
+struct ContentView: View {
+  @State private var email: String = ""
+  @State private var presentSheet = false
+
+  var body: some View {
+
+    NavigationView {
+      VStack(spacing: 50.0) {
+        Text("Authenticate users with only their email, no password required!")
+          .frame(maxWidth: .infinity, alignment: .leading)
+        
+        Spacer()
+          .frame(height: 0.0)
+        
+        CustomStyledTextField(
+          text: $email, placeholder: "Email", symbol: "person.circle.fill"
+        )
+
+        CustomStyledButton(title: "Send Sign In Link") {
+          sendSignInLink(to: email)
+        }
+
+        Spacer()
+      }
+      .padding()
+      .navigationBarTitle("Passwordless Login")
+
+    }
+    .onOpenURL { (url) in
+      let link = url.absoluteString
+      if Auth.auth().isSignIn(withEmailLink: link) {
+        passwordlessSignIn(email: email, link: link) { (success) in
+          self.presentSheet = success
+        }
+      }
+    }
```

Comment:
I did not call any Dynamic Links API's. Interestingly the above code works well regardless. If we use the dynamic link APIs to verify the incoming link is indeed a dynamic link, I think the approach below would work:

```swift
   // The above approach does not use the dynamic link APIs.
   .onOpenURL { (url) in
     DynamicLinks.dynamicLinks().handleUniversalLink(url) { (dynamicLink, error) in
       guard error == nil, let link = dynamicLink?.url?.absoluteString else {
         print(error!)
         return
       }

       if Auth.auth().isSignIn(withEmailLink: link) {
         passwordlessSignIn(email: email, link: link) { (success) in
           self.presentSheet = success
         }
       }
     }
   }
```

After chatting with DL team, it seems like calling `handleUniversalLink()` does handles reopening attribution and some analytics for the link. I'm unsure whether it is still necessary here though. Thoughts?